learn/components/styling.md

```diff
@@ -3,4 +3,309 @@ title: Styling
+  </template>
+</fancy-p>
+```
+
+The `<p>` element within the shadow tree is not effected by the styles declared outside of the shadow tree.
+
+## Inheritance
+
+Custom elements abide by the same rules of inheritance as other HTML elements. Properties such as `font-size`,
+`font-family`, and `color` are inherited from a parent element by default. So are [CSS custom properties][]. Top-level
+elements within a shadow tree inherit their inheritable properties from the custom element itself (also known as the
+shadow host). Here’s a demonstration of this at play:
+
+```html
+<style>
+  article {
+    color: deeppink;
+  }
+</style>
+
+<article>
+  <h1>This text is deeppink.</h1>
+  <article-meta>
+    <template shadowroot="open">
+      <style>
+        span {
+          font-style: italic;
+        }
+      </style>
+      <span>By Some Person</span>
+    </template>
+  </article-meta>
+</article>
+```
+
+The `<article-meta>` custom element inherits its `color` from the `<article>` element where it is set to `deeppink`. The
+`<span>` element within the shadow tree inherits its `color` from the `<article-meta>` custom element which means the
+value will be `deeppink`.
+
+## Styling elements outside of a shadow tree
+
+In order to be portable, Web Components can provide default styles for a couple types of elements which exist outside of
+its shadow tree: the custom element itself (also known as the shadow host) and slotted elements.
+
+### Writing default styles for the shadow host with `:host` and `:host()`
+
+There are two selectors which can be used to style the shadow host from within the shadow tree. They are the `:host`
+pseudo-class and the `:host()` pseudo-class function. The first will always select the shadow host. Here’s `:host` in
+action:
+
+```html
+<fancy-p>
+  <template shadowroot="open">
+    <style>
+      :host {
+        display: inline-block;
+      }
+    </style>
+  </template>
+</fancy-p>
+```
+
+The second will select the shadow host if it matches a selector. For example, given the following default styles for the
+`<fancy-p>` component, only the second element in the subsequent code example will have the styles applied.
+
+```css
+:host([extra]) {
+  font-style: italic;
+  font-weight: bold;
+}
+```
+
+```html
+<fancy-p>I not am extra</fancy-p>
+<fancy-p extra>I am extra</fancy-p>
+```
+
+#### Chaining selectors after `:host`
+
+While the `:host` selector refers to the shadow host element which is outside of the shadow tree, if you chain selectors
+it will select elements within the shadow tree.
+
+```html
+<fancy-p>
+  <template shadowroot="open">
+    <style>
+      :host > p {
+        color: deeppink;
+      }
+    </style>
+    <p>I am deeppink.</p>
+  </template>
+  <p>I am not deeppink.</p>
+</fancy-p>
+```
+
+You can `::slotted()` to style light DOM descendants of a shadow host.
+
+### Writing default styles for slotted elements with `::slotted()`
```

Comment:
One thing missing from this section is `::slotted(*)` which feels like something worth documenting.